plugins/kubernetes-common/src/permissions.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createPermission } from '@backstage/plugin-permission-common';
+
+/** This permission is used to authorize `read` actions that involve using the kubernetes Proxy Endpoint /proxy
+ * @alpha
+ */
+export const kubernetesProxyReadPermission = createPermission({
+  name: 'kubernetes.proxy.read',
+  attributes: { action: 'read' },
+});
+
+/** This permission is used to authorize `create` actions that involve using the kubernetes Proxy Endpoint /proxy
+ * @alpha
+ */
+export const kubernetesProxyCreatePermission = createPermission({
+  name: 'kubernetes.proxy.create',
+  attributes: { action: 'create' },
+});
+
+/**
+ * List of all cluster permissions.
+ * @alpha
+ */
+export const kubernetesClusterPermissions = [
+  kubernetesProxyReadPermission,
+  kubernetesProxyCreatePermission,
+];
```

Comment:
I think we only need a single permission -- the implementation doesn't do anything related to an `action` attribute.
```suggestion
/** This permission is used to check access to the proxy endpoint
 * @alpha
 */
export const kubernetesProxyPermission = createPermission({
  name: 'kubernetes.proxy',
  attributes: {},
});

/**
 * List of all Kubernetes permissions.
 * @alpha
 */
export const kubernetesPermissions = [
  kubernetesProxyPermission,
];
```

Reply:
Took your suggestion and reverted permission back to a single one.